ext/opcache/ZendAccelerator.c

```diff
@@ -1424,6 +1428,11 @@ zend_result zend_accel_invalidate(zend_string *filename, bool force)
 
 		file_handle.opened_path = NULL;
 		zend_destroy_file_handle(&file_handle);
+	} else if (!file_found) {
+		accelerator_shm_read_unlock();
+		zend_string_release_ex(realpath, 0);
+
+		return FAILURE;
```

Comment:
I think, this block is not necessary. Returning ``SUCCESS`` should be OK.

Reply:
I thought it would be nice to notify the caller that something went wrong - we haven't actually done anything. 
Do you think it's rather unnecessary?

Reply:
Returnig ``true`` won't contradict the manual. See https://www.php.net/manual/en/function.opcache-invalidate.php Returning ``false`` would require additional explanation.
Let return ``true``. Then this is just a bug fix.

@bukka Personally, I don't see problems merging this into PHP-8.2 or 8.1

Reply:
If `realpath` would modify the path then `zend_accel_invalidate` would fail. Reporting this seems nice. However, you might avoid the redundant lines by adjusting `return file_found ? SUCCESS : FAILURE;` and by adding `file_found = true;` in the `if (persistent_script && !persistent_script->corrupted) {` branch.

Reply:
Removed this branch along with the`file_found`. I will be happy to update the implementation further if someone feels like there's something to improve.